lib/unpublish.js

```diff
@@ -1,110 +1,114 @@
-/* eslint-disable standard/no-callback-literal */
 'use strict'
 
-module.exports = unpublish
-
-const BB = require('bluebird')
-
-const figgyPudding = require('figgy-pudding')
-const libaccess = require('libnpmaccess')
-const libunpub = require('libnpmpublish').unpublish
+const path = require('path')
+const util = require('util')
 const log = require('npmlog')
 const npa = require('npm-package-arg')
-const npm = require('./npm.js')
-const npmConfig = require('./config/figgy-config.js')
+const libaccess = require('libnpmaccess')
 const npmFetch = require('npm-registry-fetch')
-const otplease = require('./utils/otplease.js')
-const output = require('./utils/output.js')
-const path = require('path')
-const readJson = BB.promisify(require('read-package-json'))
+const libunpub = require('libnpmpublish').unpublish
+const readJson = util.promisify(require('read-package-json'))
+
+const npm = require('./npm.js')
 const usage = require('./utils/usage.js')
-const whoami = BB.promisify(require('./whoami.js'))
+const output = require('./utils/output.js')
+const otplease = require('./utils/otplease.js')
+const whoami = util.promisify(require('./whoami.js'))
+
+cmd.usage = usage('unpublish', 'npm unpublish [<@scope>/]<pkg>[@<version>]')
 
-unpublish.usage = usage('unpublish', 'npm unpublish [<@scope>/]<pkg>[@<version>]')
+module.exports = cmd
+function cmd(args, cb) {
+  unpublish(args, cb)
+    .then((ret) => cb(null, ret))
+    .catch((err) => err.code === 'EUSAGE' ? cb(err.message) : cb(err))
+}
 
-function UsageError () {
-  throw Object.assign(new Error(`Usage: ${unpublish.usage}`), {
-    code: 'EUSAGE'
-  })
+cmd.completion = (args, cb) => {
+  completion(args)
+    .then(() => cb())
+    .catch(cb)
 }
+async function completion(args) { 
+  const { partialWord, conf } = args
+
+  if (conf.argv.remain.length >= 3) return cb()
+
+  const username = await whoami([], true)
+  if (!username) { return [] }
+  const opts = { ...npm.flatOptions }
```

Comment:
I think this is an unnecessary clone, since you don't ever mutate the opts object.